packages/app/src/cli/services/dev/processes/dev-session.ts

```diff
@@ -0,0 +1,209 @@
+        case EventType.Updated:
+          processOptions.stdout.write(`🔄 Extension Updated ->> ${eve.extension.handle}`)
+          break
+      }
+    })
+    try {
+      await Promise.all(promises)
+      // eslint-disable-next-line no-catch-all/no-catch-all, @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      processOptions.stderr.write('Error building extensions')
+      processOptions.stderr.write(error.message)
+    }
+
+    const networkStartTime = startHRTime()
+    await performActionWithRetryAfterRecovery(async () => {
+      await bundleExtensionsAndUpload({...processOptions, app: event.app}, true)
+    }, refreshToken)
+
+    const endTime = endHRTimeInMs(event.startTime)
+    const endNetworkTime = endHRTimeInMs(networkStartTime)
+    processOptions.stdout.write(`Session updated [Network: ${endNetworkTime}ms -- Total: ${endTime}ms]`)
+  })
+
+  // Start watching for changes in the app
+  await appWatcher.start()
+  processOptions.stdout.write(`Dev session ready, watching for changes in your app`)
+}
+
+/**
+ * Build all extensions for the initial bundle
+ * All subsequent changes in extensions will trigger individual builds
+ *
+ * @param options - The options for the process
+ */
+async function initialBuild(options: DevSessionProcessOptions) {
+  const allPromises = options.app.realExtensions.map((extension) => {
+    return extension.buildForBundle(
+      {...options, app: options.app, environment: 'development'},
+      options.bundlePath,
+      undefined,
+    )
+  })
+  await Promise.all(allPromises)
+}
+
+/**
+ * Bundle all extensions and upload them to the developer platform
+ * Generate a new manifest in the bundle folder, zip it and upload it to GCS.
+ * Then create or update the dev session with the new assets URL.
+ *
+ * @param options - The options for the process
+ * @param updating - Whether the dev session is being updated or created
+ */
+async function bundleExtensionsAndUpload(options: DevSessionProcessOptions, updating: boolean) {
+  outputDebug('Bundling and uploading extensions', options.stdout)
+  const bundleZipPath = joinPath(dirname(options.bundlePath), `bundle.zip`)
+
+  // Generate app manifest in the bundle folder (overwriting the previous one)
+  const appManifest = await options.app.manifest()
+  const manifestPath = joinPath(options.bundlePath, 'manifest.json')
+  await writeFile(manifestPath, JSON.stringify(appManifest, null, 2))
+
+  // Create zip file with everything
+  await zip({
+    inputDirectory: options.bundlePath,
+    outputZipPath: bundleZipPath,
+  })
+
+  // Get a signed URL to upload the zip file
+  const signedURL = await getExtensionUploadURL(options.developerPlatformClient, {
+    apiKey: options.appId,
+    organizationId: options.organizationId,
+    id: options.appId,
+  })
+
+  // Upload the zip file
+  const form = formData()
+  const buffer = readFileSync(bundleZipPath)
+  form.append('my_upload', buffer)
+  await fetch(signedURL, {
```

Comment:
it is easy, and since the types and names are identical, the only difference is that you need to have the explicit import. A lint rule sounds like a good idea.